frontend/src/common/hooks/useCategoryFilter/common/selectUtils.ts

```diff
@@ -0,0 +1,296 @@
+  const { pinnedCategoryValues, tooltip } =
+    CATEGORY_FILTER_CONFIGS_BY_ID[categoryFilterId];
+
+  const allCategoryValueViews = buildSelectCategoryValueViews(
+    config,
+    [...categoryValueByValue.values()],
+    ontologyTermLabelsById
+  ).sort(sortCategoryValueViews("categoryValueId"));
+
+  // Split values into pinned and unpinned.
+  const [pinnedValues, unpinnedValues] = partitionSelectCategoryValueViews(
+    allCategoryValueViews,
+    pinnedCategoryValues
+  );
+
+  // Build view model of select category.
+  const selectView: SelectCategoryView = {
+    categoryFilterId: categoryFilterId,
+    label: config.label,
+    pinnedValues,
+    unpinnedValues,
+    values: allCategoryValueViews,
+  };
+
+  // Handle special cases where select category may be disabled.
+  if (
+    categoryFilterId === CATEGORY_FILTER_ID.ETHNICITY &&
+    !isEthnicityViewEnabled(filterState)
+  ) {
+    selectView.isDisabled = true;
+    selectView.tooltip = tooltip;
+  }
+  // Otherwise, check generic case where category is disabled due to no values meeting current filter.
+  else if (isSelectCategoryDisabled(selectView)) {
+    selectView.isDisabled = true;
+    selectView.tooltip = TOOLTIP_CATEGORY_DISABLED;
+  }
+
+  return selectView;
+}
+
+/**
+ * Build updated set of selected filters for the given single or multiselect category and the selected category values.
+ * @param config - Configuration model of selected category.
+ * @param selectedValue - Category value ID to toggle the selected state of.
+ * @param currentFilters - Current set of selected category values.
+ * @returns Array of selected category values for the given category.
+ */
+export function buildNextSelectCategoryFilters<T extends Categories>(
+  config: CategoryFilterConfig,
+  selectedValue: CategoryValueId,
+  currentFilters: Filters<T>
+): CategoryValueId[] {
+  const { categoryFilterId, multiselect } = config;
+
+  // Grab the current selected values for the category.
+  const categoryFilters = getCategoryFilter(categoryFilterId, currentFilters);
+
+  // Currently, no filters already selected for this category; add category value as first.
+  if (!categoryFilters) {
+    return [selectedValue];
+  }
+
+  // Create new array of selected category value keys, with the selected state of the given category value toggled.
+  return toggleCategoryValueSelected(
+    selectedValue,
+    categoryFilters.value,
+    multiselect
+  );
+}
+
+/**
+ * Build select category value view models of the given select category values.
+ * @param config - Config model of category to build category value views for.
+ * @param selectCategoryValues - Values to build view models from.
+ * @param ontologyTermLabelsById - Set of ontology term labels keyed by term ID, used to determine labels for ontology
+ */
+export function buildSelectCategoryValueViews(
+  config: CategoryFilterConfig,
+  selectCategoryValues: SelectCategoryValue[],
+  ontologyTermLabelsById: Map<string, string>
```

Comment:
Added `ontologyTermLabelsById` to allow filtering of the new ontology fields to be driven by ontology term ID while retaining the label for display.